### PR TITLE
parse multiple URLs and response format

### LINKED ISSUE #8: Made NeMo can parse more than one URL
currently if you use `{URL}` format it will be access the URL and parse the `message` key value as message, but weirdly its only working one format, I need to work for many format

### PR DESCRIPTION
**Pull request by ronaudinho**
## Description
fixes #8 
- set up multiple urls for test
- customize the response message as you see fit and edit/create supported key-format config

in my case:
- i setup local endpoint that accepts post request
- each response uses different format
  - url_1 returns `{"message": "message"}
  - url_2 returns `{"massage": "message"}
  - url_1 returns `{"mesej": "message"}
- you can follow `config/keys-example.json` for sample format and set your `SUPPORTED_KEYS_CONFIG` env to your own file

caveats:
- still only accepts JSON (I assume you only need to parse JSON response)
- custom format only looks up to first level of JSON e.g using `config/keys-example.json`, it can parse `{"mesej": "message"}` but not `{"mesej": {"message: "message"}}`. I assume that you do not need support for parsing nested JSON.
- i added custom message if request fails/format is not supported since I am not sure what you want for this, feel free to change it

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?
`go test -run TestNemoParser`, I don't use WhatsApp

## Checklist:
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Misc
say my hi to Okza, apparently he is not a member of this repo so I cannot tag him

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -1,29 +1,111 @@
...
+		return defSupportedRespKeys
+	}
+	b, err := ioutil.ReadFile(file)
+	if err != nil {
+		return defSupportedRespKeys
+	}
+	var m map[string]interface{}
+	json.Unmarshal(b, &m)
+	for k, v := range m {
+		pp, ok := v.(float64)
```
Comment (ronaudinho) [inline parser.go:104]: `json.Unmarshal` treats all number as float64, either implement your own `UnmarshalJSON` or leave it as it is

File: parser.go [hunk 2]
```diff
@@ -1,29 +1,111 @@
...
+	urlCount := strings.Count(pesan, "{{")
+	if urlCount == 0 {
+		return pesan, nil
+	}
+	for i := 0; i < urlCount; i++ {
+		url := between(pesan, "{{", "}}")
 		r, err := req.Post(url, req.BodyJSON(Sessions))
 		if err != nil {
-			return "error occured, try again", err 
+			pesan = strings.Replace(pesan, fmt.Sprintf("{{%s}}", url), errReqErr(url), -1)
```
Comment (ronaudinho) [inline parser.go:39]: if there are multiple URLs and one of the request fails, replace current URL in pesan with error message and continue. change this as you need.

File: parser.go [hunk 3]
```diff
@@ -1,29 +1,111 @@
...
-		log.Println(pesan)
+// lookupKey looks up if key exists in a map based on priority
+// return empty string if key is not supported
+// currently only supports first level key
+// might want to improve the algorithm
+// i.e return once a supported key is found with the assumption
+// that response body has commonly used keys to denote response message
+// it also does not care if the config sets multiple keys with the same
+// priority, it uses whichever is found first
+func lookupKey(m map[string]interface{}, sup map[string]int) string {
```
Comment (ronaudinho) [inline parser.go:68]: if you don't care about priority/handling multiple possible keys, change supported keys to string slice (or map[string]struct{} if you need to squeeze out performance)

**Review by k1m0ch1 — APPROVED** [review]
its pretty ok, but wonderfull work.
thank you gotta test this.

**Pull request merged**